src/System.Runtime/tests/System/Text/StringBuilderTests.netcoreapp.cs

```diff
@@ -199,5 +199,40 @@ public static void Insert_CharSpan_Invalid()
             AssertExtensions.Throws<ArgumentOutOfRangeException>("index", () => builder.Insert(builder.Length + 1, new ReadOnlySpan<char>(new char[0]))); // Index > builder.Length
             AssertExtensions.Throws<ArgumentOutOfRangeException>("requiredLength", () => builder.Insert(builder.Length, new ReadOnlySpan<char>(new char[1]))); // New length > builder.MaxCapacity
         }
+
+        public static IEnumerable<object[]> Equals_String_TestData()
+        {
+            string mediumString = new string('a', 30);
+            string largeString = new string('a', 1000);
+            string extraLargeString = new string('a', 41000); // 8000 is the maximum chunk size
+
+            var sb1 = new StringBuilder("Hello");
+            var sb2 = new StringBuilder(20).Append(mediumString);
+            var sb3 = new StringBuilder(20).Append(largeString);
+            var sb4 = new StringBuilder(20).Append(extraLargeString);
+
+            yield return new object[] { sb1, "Hello", true };
+            yield return new object[] { sb1, "Hel", false };
+            yield return new object[] { sb1, "Hellz", false };
+            yield return new object[] { sb1, "Helloz", false };
+            yield return new object[] { sb1, "", false };
+            yield return new object[] { new StringBuilder(), "", true };
+            yield return new object[] { new StringBuilder(), "Hello", false };
+            yield return new object[] { sb2, mediumString, true };
+            yield return new object[] { sb2, "H", false };
+            yield return new object[] { sb3, largeString, true };
+            yield return new object[] { sb3, "H", false };
+            yield return new object[] { sb3, new string('a', 999) + 'b', false };
+            yield return new object[] { sb4, extraLargeString, true };
+            yield return new object[] { sb4, "H", false };
+        }
+
+        [Theory]
+        [MemberData(nameof(Equals_String_TestData))]
+        public static void Equals(StringBuilder sb1, string value, bool expected)
+        {
+            Assert.Equal(expected, sb1.Equals(value));
```

Comment:
Have you verified that these tests pass with your implementation?

I do not see how `sb1.Equals(value)` can work here. StringBuilder does not have overriden Equals method, and this will call `Object.Equals`, that will always return false.

Reply:
yeah I have tested these tests with my local coreclr build. and all of theses tests are getting passed

Reply:
Could you please resolve the conflicts in this PR, so I can build it against latest CoreCLR and understand what is going on?

Reply:
> I do not see how sb1.Equals(value) can work here. StringBuilder does not have overriden Equals method, and this will call Object.Equals, that will always return false.

@jkotas, string has an implicit conversion operator to `ReadOnlySpan<char>`, so isn't `sb1.Equals(value)` calling the new `StringBuilder.Equals(ReadOnlySpan<char>)`, not `Object.Equals`?

Reply:
Yes, you are right. But then the first line is testing the exact same thing as the second line and there is no point to have both of these here. I would keep just the second line since it makes it explicit what's going on.

Reply:
I tested them without the new overload. If the new overload is not present then all the ```true ``` tests fail. (As Expected)

Reply:
This is means that this overload can chance behavior of the existing programs after recompile in interesting ways.

For example:
```
StringBuilder sb = new StringBuilder("Hello");
bool ret = sb.Equals("Hello"); // Return false today, will return true with this API.
```

It is probably fine. I cannot think of anything obviously bad. Just pointing it out in case somebody can think about a bad case.